pkg/collector/tcp.go

```diff
@@ -79,31 +82,39 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 				return
 			}
 			klog.V(2).Infof("Receiving %d bytes from %s", size, address)
-			buffBytes := make([]byte, size)
-			copy(buffBytes, buff[:size])
+			klog.V(2).Infof("Remaining bytes from last read length: %v", len(remainingBytes))
+			size = size + len(remainingBytes)
+			copiedBytes := make([]byte, size)
+			copy(copiedBytes[:len(remainingBytes)], remainingBytes)
+			copy(copiedBytes[len(remainingBytes):], buff[:size-len(remainingBytes)])
 			for size > 0 {
-				length, err := getMessageLength(bytes.NewBuffer(buffBytes))
+				bytesBuffer := bytes.NewBuffer(copiedBytes)
```

Comment:
Would it be better to have one `bytesBuffer` instead of `copiedBytes` before this for loop and keep consuming it as we always use `Decode` for consuming buffer?

Reply:
Modified the logic.